modelling/modelling.py

```diff
@@ -44,7 +44,57 @@ def model_predict(data, df, name):
     model.train(data)
     model.predict(data.X_test)
     model.print_results(data)
-
-
+    
+    return results
+def model_predict_chained(data, df, name):
+    results = []
+    
+    print("RandomForest")
+    model = RandomForest("RandomForest", data.get_embeddings(), data.get_y2())
+    model.train(data)
+    model.predict(data.X_test)
+    model.print_results(data)
+    
+    model = RandomForest("RandomForest", data.get_embeddings(), data.get_y2_y3())
+    model.train(data)
+    model.predict(data.X_test)
+    model.print_results(data)
+    
+    model = RandomForest("RandomForest", data.get_embeddings(), data.get_y2_y3_y4())
+    model.train(data)
+    model.predict(data.X_test)
+    res = model.print_results(data)
+    results.append(res)
+    
+    return results
+def model_predict_hierarchical(data, df, name):

```

Comment:
This was not asked to be implemented  (Decision 2)

Reply:
The task clearly asks to consider and implement the Hierarchical Modeling approach (Design Decision 2) in addition to the Chained Multi-outputs approach (Design Decision 1).

The `model_predict_hierarchical` function starting at line 70 is indeed implementing the Hierarchical Modeling approach as described in the task. Specifically:

1. I first train a model to predict Type 2 classes.
2. For each Type 2 class, I filter the data and train a new model to predict Type 3 classes within that Type 2 class.
3. For each Type 2 and Type 3 class combination, I again filter the data and train models to predict Type 4 classes.

This aligns with the description in the task:

> For example, a model instance (of e.g., random forest) will classify Type 2, then data from Type 3 will be filtered for each class in Type 2 and a new model instance will perform classification on that filtered data. [...] Then, at the next level, we consider each filter set A as a data set for Type 3 and will filter Type 4 under each filter similarly set A as we did for Type 2-Type 3 combination.

Reply:
Apologies @codemeasandwich I still i dont see where it asks to implement Decision Choice 2
<img width="1158" alt="Screenshot 2024-04-22 at 09 13 53" src="https://github.com/codemeasandwich/aiEmail/assets/81230135/9d652bce-be86-4b3e-8ca3-e9041466a99e">